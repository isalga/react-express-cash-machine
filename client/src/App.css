html, body, p {
  font-family: Arial, Helvetica, sans-serif;
}

.container {
  display: flex;
  flex-direction: column;
  padding: 10rem;
  max-width: 300px;
  margin: 0 auto;
}

form {
  display: flex;
  flex-direction: column;
  padding: 1rem 0rem;
}

form > div {
 padding: 1rem 0rem;
}

.result-container {
  color: green;
}

.error-container {
  color: red;
}
