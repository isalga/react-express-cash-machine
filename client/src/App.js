import React, { useRef } from 'react';
import { useRequestNotes } from './hooks';
import './App.css';

const App = () => {
  const inputEl = useRef(null);
  const [{ notes, loading, error }, setEntry] = useRequestNotes();

  const handleSubmit = (e) => {
    e.preventDefault();
    setEntry(inputEl.current.value);
  };

  return (
    <div className="container">
      <form onSubmit={handleSubmit}>
        <div>
          <label>
            Entry:
            <input ref={inputEl} type="text" name="entry" />
          </label>
        </div>
        <div>
          <input type="submit" value="Request" />
        </div>
      </form>
      {loading && (
        <p>processing...</p>
      )}
      {notes && (
        <p className="result-container">
        Result: [{`${notes}`}]
        </p>
      )}
      {error && (
        <p className="error-container">
        Error: {`${error}`}
        </p>
      )}
    </div>
  );
}

export default App;
