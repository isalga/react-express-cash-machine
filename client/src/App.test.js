import React from 'react';
import renderer, { act } from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import App from './App';

it('first load renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when submit', () => {
  const component = renderer.create(<App />);
  const submitBtn = component.root.find(el => el.type === 'input' && el.props.type === 'submit');

  act(() => {
    ReactTestUtils.Simulate.click(submitBtn);
  });

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
