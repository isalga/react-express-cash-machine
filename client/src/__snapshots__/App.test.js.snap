// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first load renders correctly 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label>
        Entry:
        <input
          name="entry"
          type="text"
        />
      </label>
    </div>
    <div>
      <input
        type="submit"
        value="Request"
      />
    </div>
  </form>
</div>
`;

exports[`renders correctly when submit 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label>
        Entry:
        <input
          name="entry"
          type="text"
        />
      </label>
    </div>
    <div>
      <input
        type="submit"
        value="Request"
      />
    </div>
  </form>
  <p>
    processing...
  </p>
</div>
`;
